llvm/lib/Target/SystemZ/SystemZISelLowering.cpp

```diff
@@ -513,11 +514,26 @@ SystemZTargetLowering::SystemZTargetLowering(const TargetMachine &TM,
   }
 
   // Handle floating-point types.
+  // Promote all f16 operations to float, with some exceptions below.
+  for (unsigned Opc = 0; Opc < ISD::BUILTIN_OP_END; ++Opc)
+    setOperationAction(Opc, MVT::f16, Promote);
+  setOperationAction(ISD::ConstantFP, MVT::f16, Expand);
+  for (MVT VT : {MVT::f32, MVT::f64, MVT::f128}) {
+    setLoadExtAction(ISD::EXTLOAD, VT, MVT::f16, Expand);
+    setTruncStoreAction(VT, MVT::f16, Expand);
+    setOperationAction(ISD::FP_EXTEND, VT, Custom);
+    setOperationAction(ISD::STRICT_FP_EXTEND, VT, Custom);
```

Comment:
Minor nit, but those two should probably be done in the loop below.

Reply:
Moved them down but not sure if they should be together (as they are treated the same), or if we should maintain the separation of the STRICT operations (-> move FP_EXTEND up)

Reply:
I don't have a strong opinion on that.  How you do it currently looks fine to me.